build/release/README.md

```diff
@@ -55,26 +83,28 @@ To publish a new patch release build, follow these steps:
 
 1. Make sure all needed PRs are merged to the release branch
 2. Check that integration tests are green (except intermittent issues)
-3. Open a PR to update the version, and merge it
+3. Open a PR to update the doc/manifest image tag versions, and merge it
+   For example: https://github.com/rook/rook/pull/13301
 4. Tag the branch:
 
     ```console
     # make sure no files are checked out locally, then proceed:
-    git checkout <branch> # e.g. release-1.10
-    git pull
+    git fetch --all
+    git checkout <branch> # e.g. release-1.12
+    git reset --hard upstream/release-1.12
```

Comment:
I changed usage of `pull` and `rebase` to `git reset --hard upstream/<branch>`, which forcefully moves all files to the desired branch without any risk of merge/rebase conflicts.